Library/Homebrew/extend/ENV/super.rb

```diff
@@ -28,8 +28,7 @@ def self.extended(base)
   def self.bin
     return unless MacOS.has_apple_developer_tools?
 
-    bin = HOMEBREW_ENV_PATH.subdirs.reject { |d| d.basename.to_s > MacOS::Xcode.version }.max
-    bin.realpath unless bin.nil?
+    HOMEBREW_ENV_PATH.subdirs.max.realpath
```

Comment:
Now that's a good question and one I've been completely overlooking! I don't have a good answer, but I guess the directory currently picked here (namely `scm`) an that only contains wrappers for `git` and `svn` doesn't make for a good platform-agnostic `superenv` either.

And given that all our true `superenv` directories are symlinks to a single directory, that further raises the question (though that's not exactly a topic for this PR), whether we need this complicated selection logic at all. I guess not. (Unless the wrapper scripts take that directory into account somehow, but a cursory search didn't reveal anything.)